File: Source/UnitTests/TestCode/AccessorRefactoringTest/Siblings.cs
namespace Test
{

	public interface IResource
	{
		String ResourceName{get;}
	}

	public abstract class AbstractFileResource : IResource
	{
		public abstract String ResourceName
		{
			get;
		}
	}

	public abstract class AbstractDBResource : IResource
	{
		public abstract String ResourceName
		{
			get;
		}
	}

	public class NetworkResource : IResource
	{
		public String ResourceName
		{
			get {return getUrl();}
		}
		public Strign getUrl()
		{
			return "http://www.mayaf.org/Janett/";
		}
	}

	public class TextFileResource : AbstractFileResource
	{
		String resourceName;
		public String ResourceName
		{
			get { return resourceName; }
		}
	}

	public class BinaryFileResource : AbstractFileResource
	{
		public String ResourceName
		{
			get { return "Binary file"; }
		}
	}
}

File: Source/UnitTests/TestCode/AccessorRefactoringTest/Siblings.java
package Test;

public interface IResource
{
	String getResourceName();
}

public abstract class AbstractFileResource implements IResource
{
	public abstract String getResourceName();
}

public abstract class AbstractDBResource implements IResource
{
	public abstract String getResourceName();
}

public class NetworkResource implements IResource
{
	public String getResourceName()
	{
		return getUrl();
	}
	public Strign getUrl()
	{
		return "http://www.mayaf.org/Janett/";
	}
}

public class TextFileResource extends AbstractFileResource
{
	String resourceName;
	public String getResourceName()
	{
		return resourceName;
	}
}

public class BinaryFileResource extends AbstractFileResource
{
	public String getResourceName()
	{
		return "Binary file";
	}
}